snapshot/git/gitfiler/cloner.go

```diff
@@ -0,0 +1,45 @@
+package gitfiler
+
+import (
+	"fmt"
+	"log"
+	"os/exec"
+
+	"github.com/scootdev/scoot/os/temp"
+	"github.com/scootdev/scoot/snapshot/git/repo"
+)
+
+// A Reference Repository is a way to clone repos locally so that the clone takes less time and disk space.
+// By passing --reference <local path> to a git clone, the clone will not copy the whole ODB but instead
```

Comment:
git clone --reference uses hardlinks

So if you:

git clone --reference a a b

You'll end up with a and b. If you du a or du b, you'll see each is the same size. But if you du a b, you'll see a is big and b is tiny.